app/assets/javascripts/govuk-frontend-local.mjs

```diff
@@ -20,6 +22,12 @@ function initAll (options) {
 
   // Find all buttons with [role=button] on the scope to enhance.
   new Button(scope).init()
+
+  var skipLinks = document.querySelectorAll('[data-module="govuk-skip-link"]')
+
+  nodeListForEach(skipLinks, function (skipLink) {
```

Comment:
not sure you need to loop over skipLinks as there should only ever be one on a page.
Where I think you might need a loop is line 24 to initialise button JS for each button?

Reply:
Wow, good spot 💯! Obvious about the skiplink 🤭 and looks like the button code slipped through the net because it still worked, just by accident. The `new Button(scope).init()` code worked because the callback it assigns to keyboard events has a guard that checks the role of the target so only executes for buttons. It still isn't meant to be used like this so I've updated the code in [this commit](https://github.com/alphagov/document-download-frontend/pull/117/commits/cb9e0213be2091b9f845e22778ce92be0b60a1dd) to apply the changes you describe.

Reply:
🥳